jina/executors/decorators.py

```diff
@@ -213,77 +220,78 @@ def batching(func: Callable[[Any], np.ndarray] = None,
     :param merge_over_axis: merge over which axis into a single result
     :param slice_on: the location of the data. When using inside a class,
             ``slice_on`` should take ``self`` into consideration.
-    :param label_on: the location of the labels. Useful for data with any kind of accompanying labels
-    :param ordinal_idx_arg: the location of the ordinal indexes argument. Needed for classes
-            where function decorated needs to know the ordinal indexes of the data in the batch
-            (Not used when label_on is used)
+    :param num_data: the number of data inside the arguments
     :return: the merged result as if run :func:`func` once on the input.
 
+    ..warning:
+        data arguments will be taken starting from ``slice_on` to ``slice_on + num_data``
+
     Example:
         .. highlight:: python
         .. code-block:: python
-
             class MemoryHungryExecutor:
 
                 @batching
                 def train(self, batch: 'numpy.ndarray', *args, **kwargs):
                     gpu_train(batch)  #: this will respect the ``batch_size`` defined as object attribute
-
+                    
                 @batching(batch_size = 64)
                 def train(self, batch: 'numpy.ndarray', *args, **kwargs):
                     gpu_train(batch)
-    """
 
+            class MultiModalExecutor:
+                
+                @batching(batch_size = 64, slice_on=[1,2])
```

Comment:
Just as a comment, there is a special batching decorator for `MultiModalExecutor` itself.

Reply:
> Hey @deepampatel ,
> 
> Thank you very much for your contribution.
> 
> As a comment, I would just recommend do not try to fit every type of batching pattern in a single function since it can become really unreadable and unmantainable. So if you feel like you need a special function for some special executor or type of input you are welcome to add it!
> 
> Great work!

@JoanFM  Thanks for the quick feedback. I agree with you on the special function for special executor class.  i will add a separate class for dictionary input type.
WDYT about having `slice_on` parameter as a `Union[int, List[int]]` instead of just `int` value in `@batching_multi_input` . Dont have any particular example in mind rn where this can be used but this thought came up when i was trying to merge both decorators.